frictionless/fields/integer.py

```diff
@@ -36,6 +36,13 @@ def value_reader(cell):
             if isinstance(cell, str):
                 if pattern:
                     cell = pattern.sub("", cell)
+                # for numeric string starting with zero for example, 001, 00, 01
+                cell_s = cell.strip()
+                has_leading_zero = (
+                    cell_s.startswith("0") and len(cell_s) > 1 and cell_s.isnumeric()
+                )
+                if self.bare_number and has_leading_zero:
```

Comment:
We can put all the leading zeroes logic under the `if self.bare_number` for performance reasons